### PR TITLE
Add static analysis to test files

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/build-info-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/goutils_test.go [hunk 1]
```diff
@@ -41,71 +41,47 @@ rsc.io/sampler:v1.3.0
...
 func TestGetProjectRoot(t *testing.T) {
 	wd, err := os.Getwd()
-	if err != nil {
-		t.Error(err)
-	}
-	defer os.Chdir(wd)
+	assert.NoError(t, err)
+	defer func() {
+		assert.NoError(t, os.Chdir(wd))
+	}()
```
Comment (yahavi) [inline utils/goutils_test.go:47]:
```suggestion
	defer assert.NoError(t, os.Chdir(wd))
```

File: utils/goutils_test.go [hunk 2]
```diff
@@ -41,71 +41,47 @@ rsc.io/sampler:v1.3.0
...
-	}
+	assert.NoError(t, err)
+	assert.Equal(t, projectRoot, root)
 
 	// CD back to the original directory.
-	if err := os.Chdir(wd); err != nil {
-		t.Error(err)
+	if !assert.NoError(t, os.Chdir(wd)) {
+		return
 	}
```
Comment (yahavi) [inline utils/goutils_test.go:65]: I think that flat code in tests is easier to understand
```suggestion
	assert.NoError(t, os.Chdir(wd))
```

File: utils/goutils_test.go [hunk 3]
```diff
@@ -120,34 +96,32 @@ func TestGetDependenciesListWithIgnoreErrors(t *testing.T) {
...
-	}
+	actual, err := GetDependenciesList(filepath.Join(goModPath), log)
+	assert.NoError(t, err)
 
 	// Since Go 1.16 'go list' command won't automatically update go.mod and go.sum.
-	// Check that we rollback changes properly.
-	newSumFileContent, _, err := getGoSum(gomodPath, log)
+	// Check that we roll back changes properly.
+	newSumFileContent, _, err := getGoSum(goModPath, log)
 	if !reflect.DeepEqual(originSumFileContent, newSumFileContent) {
```
Comment (yahavi) [inline utils/goutils_test.go:125]: Let's remove the reflection here. We can use a simple assert.Equals with the text below as the message.

**Review by yahavi — CHANGES_REQUESTED** [review]

File: utils/goutils_test.go (hunk 1 above)
Comment (RobiNino) [reply utils/goutils_test.go:47]: Why? we change back to the wd as a form of cleanup.

File: utils/goutils_test.go (hunk 2 above)
Comment (RobiNino) [reply utils/goutils_test.go:65]: It is easier to understand, but if changing dir failed here the rest of the test is irrelevant. This is used to fail the test fast

**Pull request merged**